### PR TITLE
Taskerplugin

### PR DESCRIPTION
**Pull request by joaomgcd**
# Description

Adds a Tasker plugin so that users can interact with Pocket Casts from Tasker directly, without the need to use user-unfriendly intents.

For now it's kind of just a proof of concept with an action to automatically start playing any playlist (filter) by name.
It's super easy to add new actions now though, so that users can interact with Pocket Casts from Tasker in any way they choose. :)

I'm going to personally use this so I can have a button that I can press which will automatically start playing my favorite playlist without having to go into Pocket Casts' UI.

Let me know if there are any issues :) Thanks!

# Checklist

- [x] Should this change be included in the release notes? If yes, please add a line in CHANGELOG.md
- [x] Have you tested in landscape?
- [x] Have you tested accessibility with TalkBack?
- [x] Have you tested in different themes?
- [x] Does the change work with a large display font?
- [ ] Are all the strings localized?
- [ ] Could you have written any new tests?
- [x] Did you include Compose previews with any components?

### CODE REVIEW

**Review by mchowning — COMMENTED** [review]
Thanks so much for this contribution! It's very exciting to see Tasker integration for Pocket Casts! (side note—over 10 years ago, Tasker was the very first Android app that I paid for, and I still use it today 😄 )

I left a couple of comments, but I know that this is just a first iteration, so most of them don't need to be addressed in this PR if you'd rather not. The only change I think we should definitely do before merging this is removing the copied test classes.

When you're making that change, if you'd care to add an entry to the [Changelog](https://github.com/Automattic/pocket-casts-android/blob/main/CHANGELOG.md) for this, I'd appreciate it. Wording roughly along these lines could work:

```
7.26
-----
*   New Features:
    *   Added Tasker integration for selecting filters and controlling playback.
        ([#431](https://github.com/Automattic/pocket-casts-android/pull/431)).
```

File: modules/features/taskerplugin/src/main/java/au/com/shiftyjelly/pocketcasts/taskerplugin/base/ComposableBase.kt [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+                            isSearching = false
+                            keyboardController?.hide()
+                            content.onTextChange(content.itemToString(it))
+                        }) {
+                            content.itemContent(it)
+                        }
+                    }
+                }
+            }
+            if (hasTaskerVariables) {
```
[the remark this answers is not in the document]
Comment (joaomgcd) [reply modules/features/taskerplugin/src/main/java/au/com/shiftyjelly/pocketcasts/taskerplugin/base/ComposableBase.kt]: That dropdown will show all available [local (lowercase) Tasker variables](https://tasker.joaoapps.com/userguide/en/variables.html). 🙂

If you're only seeing the **%err** and **%errmsg** variables there, that means that only those 2 variables are being used in the current task.

Hope this clarifies it!

File: modules/features/taskerplugin/src/main/java/au/com/shiftyjelly/pocketcasts/taskerplugin/base/ComposableBase.kt [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    var isSelectingTaskerVariable by remember { mutableStateOf(false) }
+    val keyboardController = LocalSoftwareKeyboardController.current
+    Box {
+
+        Row {
+            val possibleItems = content.possibleItems
+
+            val hasSuggestedItems = !possibleItems.isNullOrEmpty()
+            val hasTaskerVariables = content.taskerVariables.isNotEmpty()
+            OutlinedTextField(
```
[the remark this answers is not in the document]
Comment (joaomgcd) [reply modules/features/taskerplugin/src/main/java/au/com/shiftyjelly/pocketcasts/taskerplugin/base/ComposableBase.kt:75]: Yep, makes total sense! Changed now! Thanks!

File: modules/features/taskerplugin/src/main/java/au/com/shiftyjelly/pocketcasts/taskerplugin/base/ComposableBase.kt [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+        val itemToString: (T?) -> String = { it?.toString() ?: "" },
+        val itemContent: @Composable (T) -> Unit = { Text(text = itemToString(it)) }
+    )
+}
+
+@OptIn(ExperimentalComposeUiApi::class)
+@Composable
+fun <T> ComposableTaskerInputField(content: TaskerInputFieldState.Content<T>) {
+    var isSearching by remember { mutableStateOf(false) }
+    var isSelectingTaskerVariable by remember { mutableStateOf(false) }
```
[the remark this answers is not in the document]
Comment (joaomgcd) [reply modules/features/taskerplugin/src/main/java/au/com/shiftyjelly/pocketcasts/taskerplugin/base/ComposableBase.kt]: Ok, added a small enum class to help control this. Thanks! 👍

File: modules/features/taskerplugin/src/main/java/au/com/shiftyjelly/pocketcasts/taskerplugin/base/ComposableBase.kt [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+                                }
+                            }
+                        }
+                    }
+                }
+            )
+            if (possibleItems != null && !possibleItems.isEmpty()) {
+                DropdownMenu(
+                    expanded = isSearching,
+                    onDismissRequest = { },
```
[the remark this answers is not in the document]
Comment (joaomgcd) [reply modules/features/taskerplugin/src/main/java/au/com/shiftyjelly/pocketcasts/taskerplugin/base/ComposableBase.kt]: Yeah, I totally forgot to handle dismiss requests... 😅 Fixed, thanks!

File: modules/features/taskerplugin/src/test/java/au/com/shiftyjelly/pocketcasts/taskerplugin/viewmodel/ManualCleanupViewModelTest.kt [hunk 5]
```diff
@@ -0,0 +1,96 @@
+package au.com.shiftyjelly.pocketcasts.settings.viewmodel
```
[the remark this answers is not in the document]
Comment (joaomgcd) [reply modules/features/taskerplugin/src/test/java/au/com/shiftyjelly/pocketcasts/taskerplugin/viewmodel/ManualCleanupViewModelTest.kt]: Oops! My mistake! 😅 Removed!

**Comment by joaomgcd** [comment]
Hi Matt! Thank you very much for the review! Really appreciate all the attention to detail. 🙂

And, you know what, Tasker was also the first app I bought all those years ago as well, back when I was [not Tasker's developer](https://groups.google.com/g/tasker/c/SkCtZxnts4Y/)! 

It's awesome to be able to work on it myself now and it has the added advantage of easily being able to add stuff to it that I need myself like now, with this Pocket Casts addition! I've been using Pocket Casts for a very long time and I'm very much looking forward to being able to play a filter with a physical button in my car, without even having to turn on my phone's screen! 😁

I've gone ahead and fixed all the issues you mentioned and also added a couple new controls to the **Control Playback** action in Tasker: **Play Next In Queue** and **Skip To Time**. Hope that's ok!

Let me know if there's anything else I can do. I really can't wait for this to be available for everyone!

Thank you!

**Review by mchowning — APPROVED** [review]
Thanks for the updates. I'd like us to keep iterating on the UI, so I logged an issue for that [here](https://github.com/Automattic/pocket-casts-android/issues/471). I also tested the new actions and they all worked great, so I'll go ahead and merge this. Thanks so much for the contribution!

**Comment by joaomgcd** [comment]
Awesome! :) Glad to hear it! Thanks again!

**Pull request merged**